### PR TITLE
Update deep link docs due to the flutter_deeplinking_enabled flag default value change. 

### PR DESCRIPTION
**Pull request by hannah-hyj**
The deep link flag default value is changed in https://github.com/flutter/engine/pull/52350, and the PR is included Flutter 3.27 stable release.  

The Deep link docs should be updated to reflect the changes: 
1. If you are using flutter deep link handlers, you don't need to manually set the flutter_deeplinking_enabled flag to true anymore. 
3. If you use third-party plug-ins to handle deep links, you need to  manually set the flutter_deeplinking_enabled flag to `false`

More context: [Design doc](https://docs.google.com/document/d/1TUhaEhNdi2BUgKWQFEbOzJgmUAlLJwIAhnFfZraKgQs/edit?tab=t.0)

_Issues fixed by this PR (if any):  https://github.com/flutter/website/issues/10984 

_PRs or commits this PR depends on (if any):  https://github.com/flutter/engine/pull/52350 

## Presubmit checklist

- [ ] This PR is marked as draft with an explanation if not meant to land until a future stable release.
- [ ] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [ ] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
LGTM

File: src/content/cookbook/navigation/set-up-app-links.md [hunk 1]
```diff
@@ -99,11 +98,14 @@ It provides a simple API to handle complex routing scenarios.
     ```
    
     :::note
-    The metadata tag `flutter_deeplinking_enabled` opts
-    into Flutter's default deeplink handler.
-    If you are using the third-party plugins,
-    such as [app_links][], setting this metadata tag will
-    break these plugins. Omit this metadata tag
+    If you uses the third-party plugins to handle deep links,
```
Comment (sfshaza2) [inline src/content/cookbook/navigation/set-up-app-links.md:101]:
```suggestion
    If you use a third-party plugin to handle deep links,
```

File: src/content/cookbook/navigation/set-up-app-links.md [hunk 2]
```diff
@@ -99,11 +98,14 @@ It provides a simple API to handle complex routing scenarios.
...
+    such as [app_links][], 
+    Flutter's default deeplink handler will
+    break these plugins. 
+    
+    Add this metadata tag
```
Comment (sfshaza2) [inline src/content/cookbook/navigation/set-up-app-links.md:106]:
```suggestion
    To opt out of using Flutter's default deep link handler,
     add the following metadata tag:
```

File: src/content/cookbook/navigation/set-up-app-links.md [hunk 3]
```diff
@@ -99,11 +98,14 @@ It provides a simple API to handle complex routing scenarios.
...
-    break these plugins. Omit this metadata tag
+    If you uses the third-party plugins to handle deep links,
+    such as [app_links][], 
+    Flutter's default deeplink handler will
+    break these plugins. 
+    
+    Add this metadata tag 
+    ```<meta-data android:name="flutter_deeplinking_enabled" android:value="false" />```
+    to opt out of Flutter's default deeplink handler 
     if you prefer to use third-party plugins.
```
Comment (sfshaza2) [inline src/content/cookbook/navigation/set-up-app-links.md:109]:
```suggestion
```

File: src/content/cookbook/navigation/set-up-universal-links.md [hunk 4]
```diff
@@ -82,40 +82,15 @@ It provides a simple API to handle complex routing scenarios.
...
-
-   :::note
-   The `FlutterDeepLinkingEnabled` property enables
-   Flutter's default deeplink handler.
-   If you use a third-party plugin, such as [app_links][],
-   setting this property breaks the third-party plugin.
-   Skip this step if you prefer to use a third-party plugin.
-   :::
+  :::note
+  If you are using the third-party plugins to handle deep links,
```
Comment (sfshaza2) [inline src/content/cookbook/navigation/set-up-universal-links.md:86]:
```suggestion
  If you are use a third-party plugins to handle deep links, 
```

File: src/content/cookbook/navigation/set-up-universal-links.md [hunk 5]
```diff
@@ -82,40 +82,15 @@ It provides a simple API to handle complex routing scenarios.
...
-   Skip this step if you prefer to use a third-party plugin.
-   :::
+  :::note
+  If you are using the third-party plugins to handle deep links, 
+  such as [app_links][],
+  Flutter's default deeplink handler will
+  break these plugins. 
+
+  Add A Key and Value pair `FlutterDeepLinkingEnabled` and `NO` to info.Plist 
+  if you prefer to use third-party plugins.
```
Comment (sfshaza2) [inline src/content/cookbook/navigation/set-up-universal-links.md:92]:
```suggestion
  If you use a third-party plugin,
  add the key and value pair `FlutterDeepLinkingEnabled` and `NO` to info.Plist.
```

File: src/content/ui/navigation/deep-linking.md [hunk 6]
```diff
@@ -60,9 +60,9 @@ To get started, see our cookbooks for Android and iOS:
 If you have written a plugin to handle deep links, as described in
 [Deep Links and Flutter applications][plugin-linking]
 (a free article on Medium),
-it will continue to work until you opt in to this behavior by adding
-`FlutterDeepLinkingEnabled` to `Info.plist` or
-`flutter_deeplinking_enabled` to `AndroidManifest.xml`, respectively.
+You should opt out the default flutter deep links handler by adding
+`FlutterDeepLinkingEnabled` to false in `Info.plist` or
+`flutter_deeplinking_enabled` to false in `AndroidManifest.xml`, respectively.
```
Comment (sfshaza2) [inline src/content/ui/navigation/deep-linking.md:65]:
```suggestion
you should opt out the Flutter's default deep link handler.
To do this, set `FlutterDeepLinkingEnabled` to false in `Info.plist` _or_
`flutter_deeplinking_enabled` to false in `AndroidManifest.xml`.
```

Just double checking. You don't need to do both? It's one or the other?

**Review by sfshaza2 — COMMENTED** [review]
Thanks, @hannah-hyj, for this PR! I've done some word smithing and have one question before it can land.

File: src/content/ui/navigation/deep-linking.md (hunk 6 above)
Comment (hannah-hyj) [reply src/content/ui/navigation/deep-linking.md]: One for iOS and one for Android,  

You don't need to do both? It's one or the other? >>It depends on if you develop on both platform or just one.

**Pull request merged**